modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/internal/delete_entity_command.rb

```diff
@@ -28,37 +28,76 @@
 module Storages::Peripherals::StorageInteraction::Nextcloud::Internal
   class DeleteEntityCommand
-    UTIL = ::Storages::Peripherals::StorageInteraction::Nextcloud::Util
+    Util = ::Storages::Peripherals::StorageInteraction::Nextcloud::Util
+    Auth = ::Storages::Peripherals::StorageInteraction::Authentication
 
-    def initialize(storage)
-      @uri = storage.uri
-      @username = storage.username
-      @password = storage.password
+    def self.call(storage:, auth_strategy:, location:)
+      new(storage).call(auth_strategy:, location:)
     end
 
-    def self.call(storage:, location:)
-      new(storage).call(location:)
+    def initialize(storage)
+      @storage = storage
     end
 
-    def call(location:)
-      response = OpenProject
-                   .httpx
-                   .basic_auth(@username, @password)
-                   .delete(UTIL.join_uri_path(@uri,
-                                              "remote.php/dav/files",
-                                              CGI.escapeURIComponent(@username),
-                                              UTIL.escape_path(location)))
+    def call(auth_strategy:, location:)
+      origin_user_id = origin_user_id(auth_strategy)
+      if origin_user_id.failure?
+        return origin_user_id
+      end
+
+      response = Auth[auth_strategy].call(storage: @storage) do |http|
+        http.delete(Util.join_uri_path(@storage.uri,
+                                       "remote.php/dav/files",
+                                       CGI.escapeURIComponent(origin_user_id.result),
+                                       Util.escape_path(location)))
+      end
 
       case response
       in { status: 200..299 }
         ServiceResult.success
       in { status: 404 }
-        UTIL.error(:not_found)
+        failure(code: :not_found,
+                payload: response.json(symbolize_keys: true),
+                log_message: "Outbound request destination not found!")
       in { status: 401 }
-        UTIL.error(:unauthorized)
+        failure(code: :unauthorized,
+                payload: response.json(symbolize_keys: true),
+                log_message: "Outbound request not authorized!")
+      else
+        failure(code: :error,
+                payload: response.json(symbolize_keys: true),
+                log_message: "Outbound request failed with unknown error!")
+      end
+    end
+
+    private
+
+    def origin_user_id(auth_strategy)
+      case auth_strategy.key
+      when :basic_auth
+        ServiceResult.success(result: @storage.username)
+      when :oauth_user_token
+        origin_user_id = OAuthClientToken.find_by(user_id: user, oauth_client_id: @storage.oauth_client.id)&.origin_user_id
+        origin_user_id.present? ?
+          ServiceResult.success(result: origin_user_id) :
+          failure(code: :error,
+                  payload: nil,
+                  log_message: "No origin user ID or user token found. Cannot execute query without user context.")
```

Comment:
🟡 As this is a multi-line statement, I think it would be clearer as normal if, rather than a ternary.

Reply:
sad eric is sad :D he really likes ternaries, because they have less lines ;) 

But I get the motivation, and I understand not everyone is a ternary junkie